Assets/Global/Scripts/Player/Tail.cs

```diff
@@ -1,14 +1,31 @@
 using System;
 using UnityEngine;
-
+using System.Collections.Generic;
 public class Tail : MonoBehaviour
 {
     public Player player;
 
+    private List<Collider> colliders;
+    public void Start()
+    {
+        colliders = new List<Collider>();
+    }
+    public void Update()
+    {
+        colliders = player.tailCanDoDamage? colliders : new List<Collider>();
+    }
     public void OnTriggerEnter(Collider Collider)
     {
         if (Collider.gameObject.CompareTag("Enemy"))
         {
+            if(!colliders.Contains(Collider))
```

Comment:
can this be refactored to be an early return instead? 
`if(colliders.Contains(Collider)) return` 
would return from this method just the same as this if/else statement i think.
and would make the code a little cleaner imo.

=====
```cs
  public void OnTriggerEnter(Collider Collider)
    {
        if (Collider.gameObject.CompareTag("Enemy"))
        {
            if (colliders.Contains(Collider)) return;

            colliders.Add(Collider);

            if (player.tailCanDoDamage)
            {
                float actualDamage = player.tailDoDamage * player.playerStatistic.AttackDamageMultiplier.GetValue();
                Collider.GetComponent<EnemyBase>().OnHit((int)MathF.Round(actualDamage, 0));
            }
        }
    }
```

Reply:
You can also in the first line of the code have a guard clause for the damage.

```cs
  public void OnTriggerEnter(Collider Collider)
    {
       if (!player.tailCanDoDamage) return;
       
        if (Collider.gameObject.CompareTag("Enemy"))
        {
            if (colliders.Contains(Collider)) return;

            colliders.Add(Collider);

                float actualDamage = player.tailDoDamage * player.playerStatistic.AttackDamageMultiplier.GetValue();
                Collider.GetComponent<EnemyBase>().OnHit((int)MathF.Round(actualDamage, 0));
          
        }
    }
    ```
    
    Since so far i know we dont do anything with the trigger if you cant do damage.
    And that is 1 single boolean check. theoretically this is more performant. Even though its so little that you can ignore that fact lol